What's new June 2025 blog post (#1651)
withastro/astro.build: src/content/blog/whats-new-june-2025.mdx

import ShowcaseGridById from './_whats-new-components/ShowcaseGridById.astro';
import ThemeGrid from './_whats-new-components/ThemeGrid.astro';
import YouTubeGrid from './_whats-new-components/YouTubeGrid.astro';
import { Tweet } from 'astro-embed';
import ShowcaseCard from '~/pages/showcase/_components/ShowcaseCard.astro';
import { getEntry } from 'astro:content';
import Mention from "/src/components/Mention.astro";

Some exciting news from around the Astro ecosystem this month, including:

- [Updates from Mission Control](#updates-from-mission-control)
- [Latest releases](#latest-releases)
- [Astro Adopters](#astro-adopters)
- [Weird Wide Web](#weird-wide-web)
- [Cosmic Canvas](#cosmic-canvas)
- [Community updates](#community)
- [Astro in the news, migration stories, how-to articles, and more!](#content)
- [Watch Party](#watch-party)
- [Tips & Tools](#tips--tools)
- [Themes & Templates](#themes--templates)
- [Showcase](#showcase)
- [Starlight in the wild](#starlight-in-the-wild)

Let's dive right into it!

## Updates from Mission Control 

- 🚀 [Astro Solstice Festival](https://astro.build/solstice/)

> - 🎨 [Spirit of Astro design contest](https://shop.astro.build/contest/)

Author: ```suggestion
- 🛒 [Astro Mart](https://shop.astro.build)
- 🎨 [Spirit of Astro design contest](https://shop.astro.build/contest/)
```

## Latest releases

Minor releases introduce new features and improvements. Check out what's new!

- [Astro 5.9](https://github.com/withastro/astro/blob/refs/heads/main/packages/astro/CHANGELOG.md#590) - experimental Content Security Policy (CSP) support, rendering Markdown in content loaders, and more!
- [Astro 5.10](https://github.com/withastro/astro/blob/refs/heads/main/packages/astro/CHANGELOG.md#5100) - responsive images, experimental live content collections, and more!

## Astro Adopters

From Unilever to Cash App, some of the biggest names you know are using Astro!

export const featuredSites = [
  'www.unilever.com',
  'design.cash.app,'
]; 

{/* <ShowcaseGridById ids={featuredSites} /> */}

<div class="grid grid-cols-1 sm:grid-cols-2 gap-5">
  <ShowcaseCard site={await getEntry('showcase', 'www.unilever.com')} />

>   <ShowcaseCard site={await getEntry('showcase', 'design.cash.app')} />

Author: ```suggestion
  <ShowcaseCard site={await getEntry('showcase', 'design.cash.app')} />
  <ShowcaseCard site={await getEntry('showcase', 'pages.nist.gov')} />
  <ShowcaseCard site={await getEntry('showcase', 'openai.github.io')} />
```

</div>

## Weird Wide Web

Some sites from the last month that show off what you can do with Astro that are quirky, innovative, inspiring, or just plain fun!

- [Ascii Moon](https://asciimoon.com) - View & cycle through The Moon's phases day to day - rendered in ASCII art.

> - [Give Back Guide](https://giveback.guide/) - A platform that connects travellers with projects and experiences that have a positive impact on people, places and the planet. Their goal is to make it easy for travellers to find and support local projects in their destinations that align with their values.

Reviewer: ```suggestion
- [Give Back Guide](https://giveback.guide/) - A platform that connects travellers with projects and experiences that have a positive impact on people, places, and the planet. Their goal is to make it easy for travellers to find and support local projects in their destinations that align with their values.
```

- 🔌 [Starlight Plugins by Example](https://hideoo.dev/notebooks/starlight-plugins-by-example) is a collection of personal notes and examples from core Astro and Starlight maintainer <Mention name="hideoo" /> with everything you need to understand and build Starlight plugins!
- 🧭 See what happened when Cassidy Williams [built for a physical scavenger hunt with Astro, Tally, and Netlify](https://cassidoo.co/post/alinea-20-hunt/) for the promise of free, fancy food!
- 🏋 Learn why [Why Astro is Your Go-To Framework for Content-Heavy Websites](https://strapi.io/blog/why-astro-js-is-your-go-to-framework-for-content-heavy-websites-with-chris-from-coding-in-public) with Chris From Coding In Public.
- 🤿 Kenji took [a dive into the internal implementation of Astro’s Content Collections](https://zenn.dev/ignorant_kenji/articles/8c0236cfa7c726) to see how Markdown is rendered. (Japanese)
- 🥇 Find out why [Astro was the perfect tool to build the Bramble website](https://madebybramble.co.uk/blog/building-the-bramble-website-with-astro/).

> 

Author: ```suggestion
- 🌞 Dwane / lifelongdev gives [a run down of Solstice Festival and the new Astro swag](https://untilit.works/break/CAQAVE4Tyju3EwwdKigC) shop.
```

- [Latest releases](#latest-releases)
- [Astro Adopters](#astro-adopters)
- [Weird Wide Web](#weird-wide-web)
- [Cosmic Canvas](#cosmic-canvas)
- [Community updates](#community)
- [Astro in the news, migration stories, how-to articles, and more!](#content)
- [Watch Party](#watch-party)
- [Tips & Tools](#tips--tools)
- [Themes & Templates](#themes--templates)
- [Showcase](#showcase)
- [Starlight in the wild](#starlight-in-the-wild)

Let's dive right into it!

## Updates from Mission Control 

- 🚀 [Astro Solstice Festival](https://astro.build/solstice/)
- 🛒 [Astro Mart](https://shop.astro.build)
- 🎨 [Spirit of Astro design contest](https://shop.astro.build/contest/)

## Latest releases

Minor releases introduce new features and improvements. Check out what's new!

- [Astro 5.9](https://github.com/withastro/astro/blob/refs/heads/main/packages/astro/CHANGELOG.md#590) - experimental Content Security Policy (CSP) support, rendering Markdown in content loaders, and more!
- [Astro 5.10](https://github.com/withastro/astro/blob/refs/heads/main/packages/astro/CHANGELOG.md#5100) - responsive images, experimental live content collections, and more!

## Astro Adopters

> From Unilever to Cash App, to the US government to OpenAI, some of the biggest names you know are using Astro!

Reviewer 2: from/to, from/to? Or maybe needs another reformulation?

```suggestion
From Unilever to Cash App, from the US government to OpenAI, some of the biggest names you know are using Astro!
```

Reviewer 2: e.g.
```suggestion
From Unilever to Cash App, the US government, and OpenAI, some of the biggest names you know are using Astro!
```

Author: Ah, here's what we did last month:

> From Ricola to Google, Bootstrap to Visa, some of the biggest names you know are using Astro!

> From Unilever to Cash App, to the US government to OpenAI, some of the biggest names you know are using Astro!

Author: ```suggestion
From Unilever to Cash App, the US government to OpenAI, some of the biggest names you know are using Astro!
```

- 🔌 [Starlight Plugins by Example](https://hideoo.dev/notebooks/starlight-plugins-by-example) is a collection of personal notes and examples from core Astro and Starlight maintainer <Mention name="hideoo" /> with everything you need to understand and build Starlight plugins!
- 🧭 See what happened when Cassidy Williams [built for a physical scavenger hunt with Astro, Tally, and Netlify](https://cassidoo.co/post/alinea-20-hunt/) for the promise of free, fancy food!
- 🏋 Learn why [Why Astro is Your Go-To Framework for Content-Heavy Websites](https://strapi.io/blog/why-astro-js-is-your-go-to-framework-for-content-heavy-websites-with-chris-from-coding-in-public) with Chris From Coding In Public.
- 🤿 Kenji took [a dive into the internal implementation of Astro’s Content Collections](https://zenn.dev/ignorant_kenji/articles/8c0236cfa7c726) to see how Markdown is rendered. (Japanese)
- 🥇 Find out why [Astro was the perfect tool to build the Bramble website](https://madebybramble.co.uk/blog/building-the-bramble-website-with-astro/).

> - 🌞 Dwane / lifelongdev gives [a run down of Solstice Festival and the new Astro swag](https://untilit.works/break/CAQAVE4Tyju3EwwdKigC) shop.

Author: ```suggestion
- 🛍 Dwane / lifelongdev [takes you through the new Astro swag](https://untilit.works/break/CAQAVE4Tyju3EwwdKigC) shop and gives a run down of Solstice Festival events.
```

    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
  ]}
/> 

### Live coding streams

Watch people build, experiment, and learn!

<YouTubeGrid
  videos={[
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
    { id: "", title: ""},
  ]}
/> 

## Tips & Tools

The latest community-built utilities and integrations to help you build with Astro. 🛠

> - [Astro Strapi loader](https://github.com/VirtusLab-Open-Source/astro-strapi-loader) - Enables easy data loading from the Strapi Content API.

Reviewer 2: Already included below! (With a slightly better description?)

```suggestion
```

- [Astro Validate Env](https://github.com/adamhl8/astro-validate-env) - A environment variable validation integration for Astro that gives you control over when/where to validate environment variables and works in all contexts.
- [Astro Medium loader](https://github.com/ykocaman/astro-medium-loader) - An Astro loader that fetches and parses Medium RSS feeds by username. Supports canonical links, optional caching, and flexible storage strategies.
- [Astro content loader](https://github.com/dnlzro/astro-content-loader) - Load `.astro` files with the Astro Content Layer API, just as you would other formats (e.g. Markdown, MDX).
- [unDocs](https://github.com/K0Rhq/un) - A documentation framework built with Astro where you can build your own docs and component playground where you control the layout, components, and design.
- [Stripe Astro loader](https://github.com/notrab/stripe-astro-loader) - Fetch data from the Stripe API and use it in Astro collections.
- [Astro Payload local loader](https://github.com/jliguori1312/payload-astro-template/tree/main/astro-payload-local-loader) - Load and cache data from a Payload CMS instance during your Astro build, using the local API.

> - [Astro Markdown prerender](https://github.com/johnzanussi/astro-markdown-prerender) - An Astro integration that automatically sets r`oute.prerender` based on frontmatter properties in your Markdown files.

Reviewer 2: backtick typo

```suggestion
- [Astro Markdown prerender](https://github.com/johnzanussi/astro-markdown-prerender) - An Astro integration that automatically sets `route.prerender` based on frontmatter properties in your Markdown files.
```

- [Astro Loader Hashnode](https://github.com/raisedadead/astro-loader-hashnode) - A powerful content loader for integrating Hashnode blog posts into your Astro website using the Content Layer API.
- [Astro TOC generator](https://github.com/bhargav-bkpatel/astro-toc-generator) - A table of contents generator for Astro.js that works specifically with Sanity CMS blog content.
- [Astro Swiper](https://github.com/pascal-brand38/astro-swiper) - Astro component for Swiper, dedicated to slider / carousel / photo swiper / slide.
- [Astro Medium loader](https://github.com/ykocaman/astro-medium-loader) - An Astro loader that fetches and parses Medium RSS feeds by username.

> - [Astro llms text](https://github.com/4hse/astro-llms-txt) - An Astro integration to generate AI‑friendly documentation files.

Reviewer 2: Capitalisation?

```suggestion
- [Astro LLMs Text](https://github.com/4hse/astro-llms-txt) - An Astro integration to generate AI‑friendly documentation files.
```
